src/Tests/Microsoft.NET.Perf.Tests/BuildPerf.cs

```diff
@@ -0,0 +1,242 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Xml.Linq;
+using FluentAssertions;
+using Microsoft.NET.TestFramework;
+using Microsoft.NET.TestFramework.Assertions;
+using Microsoft.NET.TestFramework.Commands;
+using Microsoft.NET.TestFramework.ProjectConstruction;
+using Microsoft.Xunit.Performance.Api;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.NET.Perf.Tests
+{
+    public class BuildPerf : SdkTest
+    {
+        public BuildPerf(ITestOutputHelper log) : base(log)
+        {
+        }
+
+        //  These tests are currently disabled for full framework MSBuild because the CI machines don't
+        //  have an MSBuild that supports the /restore command-line argument
+        //  Also, Microsoft.Xunit.Performance.Api.ProcessExitInfo doesn't handle non-Windows
+        //  information gathering - disabling for non-Windows
+        [CoreMSBuildAndWindowsOnlyTheory]
+        [InlineData(ProjectPerfOperation.CleanBuild)]
+        [InlineData(ProjectPerfOperation.BuildWithNoChanges)]
+        public void BuildNetCore2App(ProjectPerfOperation operation)
+        {
+            var testProject = new TestProject()
+            {
+                Name = "NetCoreApp",
+                TargetFrameworks = "netcoreapp2.0",
+                IsSdkProject = true,
+                IsExe = true
+            };
+
+            var testAsset = _testAssetsManager.CreateTestProject(testProject, identifier: operation.ToString());
+
+            TestProject(testAsset.Path, ".NET Core 2 Console App", operation);
```

Comment:
This name, TestProject, does it mean go run a test on the project or is it a more of a name to create TestProject, which happens to do things like build.

The name seems strange here. Does not read like an action. At least, I didn't read it as such at first.